tests/Filter/FilterTestCase.php

```diff
@@ -18,17 +18,44 @@
 use Doctrine\ORM\Query\Expr\Orx;
 use Doctrine\ORM\QueryBuilder;
 use PHPUnit\Framework\TestCase;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
+
+class TestQueryBuilder extends QueryBuilder
+{
+    /** @var string[] */
+    public $query = [];
+
+    /** @var string[] */
+    public $queryParameters = [];
+}
 
 class FilterTestCase extends TestCase
 {
-    public function createQueryBuilderStub(): QueryBuilder
+    public function assertSameQuery(array $expected, ProxyQuery $proxyQuery): void
     {
-        $testCase = $this;
+        $queryBuilder = $proxyQuery->getQueryBuilder();
+        if (!$queryBuilder instanceof TestQueryBuilder) {
+            throw new \InvalidArgumentException('The query builder should be build with "createQueryBuilderStub()".');
+        }
+
+        $this->assertSame($expected, $queryBuilder->query);
+    }
 
-        $queryBuilder = $this->createStub(QueryBuilder::class);
+    public function assertSameQueryParameters(array $expected, ProxyQuery $proxyQuery): void
+    {
+        $queryBuilder = $proxyQuery->getQueryBuilder();
+        if (!$queryBuilder instanceof TestQueryBuilder) {
+            throw new \InvalidArgumentException('The query builder should be build with "createQueryBuilderStub()".');
+        }
+
+        $this->assertSame($expected, $queryBuilder->queryParameters);
+    }
+
+    public function createQueryBuilderStub(): TestQueryBuilder
+    {
+        $testCase = $this;
 
```

Comment:
We can use `$this`.
```suggestion
```

Reply:
1) Sonata\DoctrineORMAdminBundle\Tests\Filter\BooleanFilterTest::testFilterArray
Error: Using $this when not in object context

Reply:
Did you remove the `static` keyword in the lambda where this variable is used?

Reply:
I'd like to use the changes in the tests as base for the checks I need to add at #1358.
@VincentLanglet, could you please confirm if this variable can be removed?